### DISCUSSION TITLE
How to modify the data ?

### DISCUSSION THREAD

**Discussion by rkenko** [pyvista/pyvista#2757] [Q&A] [answered]
Hi, 

I am new on Git and on Pyvista. I wonder how i could : 
1. correctly import my 3D data (i have it in .npy and .vtk, i don't know which one is the best for pyvista)
2. modify the morphology of my data, not the values inside. For instance, imagine i want to compress a cube in one side and expand it in another. the cube is just the global bounding box, but it contains segmented phases, with particular morphologies
3. convert back the resulting modified data correctly in numpy array. I want to be able to see the modified phases compressed along one direction and expanded along the other one...

Please could you help me with the 3 steps ? I didn't manage to do this for the moment

Thanks

**Comment by tkoyama010** [comment, marked as answer]
1. You can do it using [pyvista.read](https://docs.pyvista.org/api/utilities/_autosummary/pyvista.read.html).
2. You can do it using [pyvista.DataSet.points](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.points.html).
3. You can also get points NumPy array using [pyvista.DataSet.points](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.points.html).

**Reply by rkenko** [reply]
Hi @tkoyama010, and thank your for your quick answer. I manage to do that with your solution. 

I have another last problem. I have an unstructured grid and i want to generate a structured grid with the values in the first unstructured grid. I think there is at least 2 steps: 

1. create the structured grid with the dimensions and spacing copied from the unstructured grid
2. interpolate the values from the unstructured grid to the structured one 

Please do you know how i can do that ?  here is my code for the moment  (i am really new to pyvista) :  

```python
import numpy as np
import pyvista as pv
import matplotlib.pyplot as plt  
# import the data
data = pv.read('result_inference.vtk')
# transform the generated uniformgrid to structured grid 
data = data.cast_to_structured_grid()
# apply operations on the grid
data.points[:,0] = data.points[:,0]/2
data.points[:,1] = data.points[:,1]*np.sqrt(2)
data.points[:,2] = data.points[:,2]*np.sqrt(2) 
# plot the resulting grid
data.plot(scalars='values',background="white",show_edges=False,show_grid=True,show_axes=True,eye_dome_lighting=True,volume=False)
# save the data
data.save('result_forged.vtk')
```

**Reply by tkoyama010** [reply]
1. You can make a structured grid using [pyvista.StructuredGrid](https://docs.pyvista.org/api/core/_autosummary/pyvista.StructuredGrid.html#pyvista.StructuredGrid).
2. You can interpolate value using [pyvista.DataSet.interpolate](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.interpolate.html#interpolate).

**Reply by rkenko** [reply]
Thank you @tkoyama010, it works, but in fact i wanted to transfer the values from the Structuredgrid to an UniformGrid.

I have created the UniformGrid using : 
```python
uniform_grid = pv.UniformGrid(dims=(int(data.bounds[1]+1),int(data.bounds[3]+1),int(data.bounds[5]+1)))
```
now i want to transfer the values from the StructuredGrid called "data" in the line of code above to the UniformGrid. The interpolate solution is not suitable for my case since my field contains binary values made of zeros and ones. Do you know another way to ensure to have binary values in the UniformGrid made from the closest point in the StructuredGrid for example ?

**Reply by tkoyama010** [reply]
The [points](https://dev.pyvista.org/api/core/_autosummary/pyvista.DataSet.points.html#pyvista.DataSet.points) method can be used to obtain the coordinates of points contained in a mesh. You may then sort the data accordingly.
